File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostSettingsTagsFragment.java
package org.wordpress.android.ui.posts;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import org.wordpress.android.R;
import org.wordpress.android.WordPress;
import org.wordpress.android.fluxc.model.SiteModel;

import static org.wordpress.android.ui.posts.PostSettingsTagsActivity.KEY_TAGS;

public class PostSettingsTagsFragment extends TagsFragment {
    public static final String TAG = "post_settings_tags_fragment_tag";

    public static PostSettingsTagsFragment newInstance(@NonNull SiteModel site, @Nullable String tags) {
        Bundle args = new Bundle();
        args.putSerializable(WordPress.SITE, site);
        args.putString(KEY_TAGS, tags);

        PostSettingsTagsFragment fragment = new PostSettingsTagsFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override protected int getContentLayout() {
        return R.layout.fragment_post_settings_tags;
    }

    @Override public void onAttach(@NonNull Context context) {
        super.onAttach(context);

        if (context instanceof PostSettingsTagsActivity) {
            mTagsSelectedListener = (TagsSelectedListener) context;
        } else {
            throw new RuntimeException(context.toString() + " must implement TagsSelectedListener");
        }
    }
}

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PostSettingsTagsFragment.kt
package org.wordpress.android.ui.posts

import android.content.Context
import android.os.Bundle
import org.wordpress.android.R
import org.wordpress.android.WordPress
import org.wordpress.android.fluxc.model.SiteModel

class PostSettingsTagsFragment : TagsFragment() {
    override fun getContentLayout() = R.layout.fragment_post_settings_tags

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        (requireActivity().application as WordPress).component().inject(this)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mTagsSelectedListener = if (context is PostSettingsTagsActivity) {
            context
        } else {
            throw RuntimeException("$context must implement TagsSelectedListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mTagsSelectedListener = null
    }

    override fun getTagsFromEditPostRepositoryOrArguments() = arguments?.getString(PostSettingsTagsActivity.KEY_TAGS)

    companion object {
        const val TAG = "post_settings_tags_fragment_tag"
        @JvmStatic fun newInstance(site: SiteModel, tags: String?): PostSettingsTagsFragment {
            val bundle = Bundle().apply {
                putSerializable(WordPress.SITE, site)
                putString(PostSettingsTagsActivity.KEY_TAGS, tags)
            }
            return PostSettingsTagsFragment().apply { arguments = bundle }
        }
    }
}
